src/account/SemiModularAccount.sol

```diff
@@ -0,0 +1,180 @@
+    ) internal override returns (uint256) {
+        if (userOpValidationFunction.eq(_FALLBACK_VALIDATION)) {
+            address fallbackSigner = _getFallbackSigner();
+
+            if (
+                SignatureChecker.isValidSignatureNow(
+                    fallbackSigner, userOpHash.toEthSignedMessageHash(), userOp.signature
+                )
+            ) {
+                return _SIG_VALIDATION_PASSED;
+            }
+            return _SIG_VALIDATION_FAILED;
+        }
+
+        return super._execUserOpValidation(userOpValidationFunction, userOp, userOpHash);
+    }
+
+    function _execRuntimeValidation(
+        ModuleEntity runtimeValidationFunction,
+        bytes calldata callData,
+        bytes calldata authorization
+    ) internal override {
+        if (runtimeValidationFunction.eq(_FALLBACK_VALIDATION)) {
+            address fallbackSigner = _getFallbackSigner();
+
+            if (msg.sender != fallbackSigner) {
+                revert FallbackSignerMismatch();
+            }
+            return;
+        }
+        super._execRuntimeValidation(runtimeValidationFunction, callData, authorization);
+    }
+
+    function _exec1271Validation(ModuleEntity sigValidation, bytes32 hash, bytes calldata signature)
+        internal
+        view
+        override
+        returns (bytes4)
+    {
+        if (sigValidation.eq(_FALLBACK_VALIDATION)) {
+            address fallbackSigner = _getFallbackSigner();
+
+            if (SignatureChecker.isValidSignatureNow(fallbackSigner, hash, signature)) {
+                return _1271_MAGIC_VALUE;
+            }
+            return _1271_INVALID;
+        }
+        return super._exec1271Validation(sigValidation, hash, signature);
+    }
+
+    function _globalValidationAllowed(bytes4 selector) internal view override returns (bool) {
+        return selector == this.updateFallbackSigner.selector || super._globalValidationAllowed(selector);
+    }
+
+    function _isValidationGlobal(ModuleEntity validationFunction) internal view override returns (bool) {
+        return validationFunction.eq(_FALLBACK_VALIDATION) || super._isValidationGlobal(validationFunction);
+    }
+
+    function _getFallbackSigner() internal view returns (address) {
+        SemiModularAccountStorage storage _storage = _getSemiModularAccountStorage();
+
+        if (_storage.fallbackSignerDisabled) {
+            revert FallbackSignerDisabled();
+        }
+
+        return _retrieveFallbackSignerUnchecked(_storage);
+    }
+
+    function _retrieveFallbackSignerUnchecked(SemiModularAccountStorage storage _storage)
+        internal
+        view
+        returns (address)
+    {
+        address storageFallbackSigner = _storage.fallbackSigner;
+        if (storageFallbackSigner != address(0)) {
+            return storageFallbackSigner;
+        }
+
+        bytes memory appendedData = LibClone.argsOnERC1967(address(this), 0, 20);
+
+        return address(uint160(bytes20(appendedData)));
+    }
```

Comment:
If this function is only used once, should we inline it within `_getFallbackSigner`?

Reply:
It was inlined before-- I extracted it to reuse the code for `getFallbackSigner()` in cases where the fallback signer is disabled, but the signer is still set. Otherwise, the getter reverts, but I'm not set on keeping it this way, wdyt?

Reply:
I might be missing some context - is the function `_retrieveFallbackSignerUnchecked` used more than once?

Reply:
Oh wait, I think I follow, but it looks like the external `getFallbackSigner()` uses the checked version of `_getFallbackSigner()`, so it will revert. Should that be changed to the unchecked one?